src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs

```diff
@@ -94,6 +100,27 @@ public static bool IsChildOfDirectory(string dir, string candidate)
             candidate = Path.GetFullPath(candidate);
             return candidate.StartsWith(dir, StringComparison.OrdinalIgnoreCase);
         }
+
+        public static bool HasTrailingValidDirectorySeparator(string path)
```

Comment:
consider removing Valid from the name here. I wasn't sure if this was a positive or negative check until I read through the code fully.